great_expectations/execution_engine/execution_engine.py

```diff
@@ -102,6 +106,27 @@ def metric_suffix(self) -> str:
         return ""
 
 
+@dataclass(frozen=True)
+class MetricComputationConfiguration(DictDot):
+    """
+    MetricComputationConfiguration is a "dataclass" object, which holds components required for metric computation.
+    """
+
+    metric_configuration: MetricConfiguration
+    metric_fn: Any
+    metric_provider_kwargs: dict
+    compute_domain_kwargs: Optional[dict] = None
+    accessor_domain_kwargs: Optional[dict] = None
+
+    def to_dict(self) -> dict:
```

Comment:
@alexsherstinsky 
Are these `to_dict()` and `to_json_dict` naming conventions already established or are you introducing this with this PR.

Reply:
@Kilo59 The latter is the interface we have had for a long time.  The former is "common in the codebase" (please see `DictDot`).  Thanks.